src/OrchardCore/OrchardCore.Recipes.Core/ConfigurationMethodProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Scripting;
+
+
+namespace OrchardCore.Recipes
+{
+    public class ConfigurationMethodProvider : IGlobalMethodProvider
+    {
+        private readonly GlobalMethod _globalMethod;
+
+        public ConfigurationMethodProvider(ShellSettings settings)
+        {
+            _globalMethod = new GlobalMethod
+            {
+                Name = "configuration",
+                Method = serviceprovider => (Func<string, object>)(name =>
+                {
+                    return settings.ShellConfiguration[name];
+                })
+            };
+        }
+
+        public IEnumerable<GlobalMethod> GetMethods()
```

Comment:
This can not be done as the return value is an `IEnumerable<GlobalMethod>` and `_globalMethod` is `GlobalMethod`.

Reply:
Seems I miss `yield return`

Reply:
@hishamco I don't think you are allowed to use `yield` on inline expression.

If you know how to do it, please let me know because I haven't found a way of making what you are suggesting to work.